pandas/tests/copy_view/test_indexing.py

```diff
@@ -794,22 +900,31 @@ def test_column_as_series(using_copy_on_write, using_array_manager):
         tm.assert_frame_equal(df, df_orig)
 
 
-def test_column_as_series_set_with_upcast(using_copy_on_write, using_array_manager):
+def test_column_as_series_set_with_upcast(
+    backend, using_copy_on_write, using_array_manager
+):
     # Case: selecting a single column now also uses Copy-on-Write -> when
     # setting a value causes an upcast, we don't need to update the parent
     # DataFrame through the cache mechanism
+    dtype_backend, DataFrame, Series = backend
     df = DataFrame({"a": [1, 2, 3], "b": [4, 5, 6], "c": [0.1, 0.2, 0.3]})
     df_orig = df.copy()
 
     s = df["a"]
-    if using_copy_on_write or using_array_manager:
+    if dtype_backend == "nullable":
+        with pytest.raises(TypeError, match="Invalid value"):
+            s[0] = "foo"
+    elif using_copy_on_write or using_array_manager:
         s[0] = "foo"
     else:
         with pd.option_context("chained_assignment", "warn"):
             with tm.assert_produces_warning(SettingWithCopyWarning):
                 s[0] = "foo"
 
-    expected = Series(["foo", 2, 3], dtype=object, name="a")
+    if dtype_backend == "nullable":
```

Comment:
Can we define expected above as well? Makes following the code paths easier

Reply:
Then I would need to define it three times instead of two times (the expected is the same for the elif and else cases), but that's fine for me if that's what you intended.

Reply:
Yep,

In my experience a test gets exponentially harder to understand, if the code paths diverge (personal opinion)